File: WebContent/ddd/simple/view/courseDesign/js/stuCourseProgress/StuCourseProgressService.js

define(["app"], function (app) {
	app.factory("StuCourseProgressService", function($resource) {
		
		var stuCourseProgressService = $resource("../DD/StuCourseProgress/:action", {});
		stuCourseProgressService.saveStuCourseProgress=function(stuCourseProgress,sucesscb,errorcb)	{
			stuCourseProgressService.save({action:"saveStuCourseProgress"},{stuCourseProgress:stuCourseProgress},sucesscb,errorcb);
		};
		
		stuCourseProgressService.deleteStuCourseProgress=function(stuCourseProgressId,sucesscb,errorcb)	{
			stuCourseProgressService.save({action:"deleteStuCourseProgress"},{stuCourseProgressId:stuCourseProgressId},sucesscb,errorcb);
		};
		
		stuCourseProgressService.updateStuCourseProgress=function(stuCourseProgress,sucesscb,errorcb) {
			stuCourseProgressService.save({action:"updateStuCourseProgress"},{stuCourseProgress:stuCourseProgress},sucesscb,errorcb);
		};
		 
		stuCourseProgressService.findStuCourseProgressById=function(stuCourseProgressId,sucesscb,errorcb) {
			stuCourseProgressService.get({action:"findStuCourseProgressById",stuCourseProgressId:stuCourseProgressId},sucesscb,errorcb);
		};
		
		stuCourseProgressService.findAllStuCourseProgress=function(sucesscb,errorcb) {
			stuCourseProgressService.query({action:"findAllStuCourseProgress"},sucesscb,errorcb);
		}
		return stuCourseProgressService;
})});

File: src/ddd/simple/service/courseDesign/StuCourseProgressService.java
package ddd.simple.service.courseDesign;

import ddd.base.persistence.EntitySet;
import ddd.simple.service.base.BaseServiceInterface;
import ddd.simple.entity.courseDesign.StuCourseProgress;

public interface StuCourseProgressService extends BaseServiceInterface {
	public StuCourseProgress saveStuCourseProgress(StuCourseProgress stuCourseProgress) ;
	
	public int deleteStuCourseProgress(Long stuCourseProgressId) ;
	
	public StuCourseProgress updateStuCourseProgress(StuCourseProgress stuCourseProgress) ;
	
	public StuCourseProgress findStuCourseProgressById(Long stuCourseProgressId) ;
	
	public EntitySet<StuCourseProgress> findAllStuCourseProgress() ;
 
}